src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -476,6 +493,36 @@ static bool IsCapturedPrimaryConstructorParameter(
             }
         }
 
+        private static ImmutableArray<ISymbol> FilterOutUncapturableParameters(ImmutableArray<ISymbol> symbols, SyntaxNode contextNode)
+        {
+            // Can't capture parameters across a static lambda/local function
+
+            var containingStaticFunctions = contextNode.AncestorsAndSelf().Where(a => a switch
+            {
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.Modifiers.Any(SyntaxKind.StaticKeyword),
+                LocalFunctionStatementSyntax localFunction => localFunction.Modifiers.Any(SyntaxKind.StaticKeyword),
+                _ => false,
+            }).ToSet();
+
+            if (containingStaticFunctions.Count == 0)
+                return symbols;
+
+            return symbols.WhereAsArray(s =>
+            {
+                if (s is not IParameterSymbol { DeclaringSyntaxReferences: [var parameterReference] })
+                    return true;
+
+                foreach (var staticFunction in containingStaticFunctions)
```

Comment:
>foreach (var staticFunction in containingStaticFunctions) [](http://example.com/codeflow?start=16&length=57)

This isn't clear to me. Is this just testing whether the symbol is a parameter defined before the last static method parent? Could the AncestorsAndSelf just keep the first containing static method?

Reply:
yes... i suppose that's sufficient. :)  will adjst

Reply:
done.

Reply:
done